OloEditor/src/Panels/AnimationGraphEditorPanel.cpp

```
@@ -114,12 +115,35 @@ namespace OloEngine
+        // serializer already canonicalises the shape we care about for undo.
+        // Cost is bounded by graph size and only fires on widget release.
+        if (Ref<AnimationGraph> currentGraph = SnapshotGraph(); currentGraph && m_EditSessionSnapshot)
+        {
+            const std::string before = AnimationGraphSerializer::SerializeToString(m_EditSessionSnapshot);
+            const std::string after = AnimationGraphSerializer::SerializeToString(currentGraph);
+            if (before == after)
+            {
+                m_EditSessionSnapshot = nullptr;
+                m_EditSessionActive = false;
+                return;
+            }
```

Cppcheck (2.20.0)

[style] The function 'CanUndo' is never used.

(unusedFunction)